dali/operators/imgcodec/image_decoder.h

```diff
@@ -597,8 +615,17 @@ class ImageDecoder : public StatelessOperator<Backend> {
       st.orig_img_type = DALI_GRAY;
       nchannels = 1;
     } else {
-      st.image_info.sample_format = NVIMGCODEC_SAMPLEFORMAT_I_RGB;
-      st.orig_img_type = DALI_RGB;
+      // TIFF decoder has some issues with decoding RGB outputs on 1-channel inputs on nvImageCodec
+      // 0.2.0 In that case, decode directly to grayscale and convert later
+      if (!version_at_least(0, 3, 0) &&
+          st.parsed_sample.nvimgcodec_img_info.sample_format == NVIMGCODEC_SAMPLEFORMAT_P_Y) {
+        st.image_info.sample_format = NVIMGCODEC_SAMPLEFORMAT_P_Y;
+        st.orig_img_type = DALI_GRAY;
+        decode_shape[2] = 1;
```

Comment:
nitpick

Wouldn't the direct st.out_shape write be more readable? It looks like we need the `decode_shape` only in this one place in the older version fallback. 

```suggestion
        st.out_shape[2] = 1;
```